### DISCUSSION TITLE
Add support for customizing scope and hierarchy

### DISCUSSION THREAD

**Discussion by benanhalt** [specify/specify7#5513] [Feature Requests]
The scoping of various records to the collection -> discipline -> division -> institution hierarchy may not be appropriate for all use cases.  What possibility is there for adding flexibility to this system?

From Martin Stein:

> When we were in Lawrence, we brought up that a more flexibly hierarchy than Institution > Division > Discipline > Collection would be great; another user definable tree, like currently for Geography, Taxonomy, Storage, etc. Than add the ability for users to define at what level certain attributes for a collection (or what ever the terminal of that a tree would be) should be applied. E.g. I sit in a local museum in the Canadian Rockies and have a specially themed invertebrate paleo collection (let's say I have got a lot of Burgess Shale material), and I can now decide that I want my localities and paleo contexts private to that collection, but the taxonomy tree accessible by the entire Section of Invertebrate Paleontology (which does not mean that all collection in that Section must use it, the collection of unique Permian sponges can use its own) the Geography tree should be for our Institution (because all our stuff is from localities nearby), while we share Agents with our Earth Science division (assuming our museum is organised Institution > Division > Section > Collection). What is shared/scoped at what level should be user definable like the tree. 

---
**@grantfitzsimmons edit:**

The most common request is to have Geography at the institution level.

**Comment by grantfitzsimmons** [comment]
Received a request to implement the Geography tree at the institutional level. Trees should have their scope be more configurable.

**Comment by maxpatiiuk** [comment]
Two solutions are proposed in https://github.com/specify/specify7/issues/2912#issuecomment-1406828873

(see "Simple" and "Advanced" sections)

In either case, there is very little we can do until sp7 divorses from sp6

**Comment by grantfitzsimmons** [comment]
From [Barcelona on Trello](https://trello.com/c/md3AMCxO/45-sharing-goegraphy-thesaurus-with-all-the-disciplines):

>No matter how much I look at it, I see no adavantage in not sharing the Geography Thesaurus with the three disciplines. Our old CMS shares all the three thesaurus and it would be better if Geography, when will begin the migration of the next collection, could be shared.
>
>Would it be possible?

**Comment by grantfitzsimmons** [comment]
Soraya at Naturhistorisches Museum Bern said the following:
>We would like to have the Geography tree scoped at the institution level. For us having it at the discipline level is not needed, and actually creates more work, because edits to the tree have to be repeated for all disciplines.

**Comment by grantfitzsimmons** [comment]
Soraya brings up a good question– `IsSingleGeographyTree` is a boolean required field in the `institution` table, but as far as I understand this was never (fully) implemented since Geography is always scoped at the discipline level.

@benanhalt Do you know anything about this? 🔍

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/geography-tree-at-institution-level/1515/2

**Comment by benanhalt** [comment]
@grantfitzsimmons I don't know anything about it. [Searching the Specify 6 code](https://github.com/search?q=repo%3Aspecify%2Fspecify6%20issinglegeographytree&type=code), the field doesn't appear to be used for anything.

**Comment by grantfitzsimmons** [comment]
@benanhalt Thanks for looking into that! I couldn't find anything on it either. It's been a while 😉😄
